### PR TITLE
Run nam tests in CI by providing local httpbin

### PR DESCRIPTION
**Pull request by nilsnolde**
ref #42554 

## Description

It doesn't fix the above issue, I couldn't really figure out how to debug the `finished()` signal.. I simply added the httpbin.org project as a service to the tests CI runner so that the `testqgsnetworkaccessmanager.cpp` tests can run. Since I didn't fix the issue, I commented out the failing parts of the tests and left TODOs. 

If someone has a hint how to track & fix the bug in #42554, I'd appreciate it!

### CODE REVIEW

File: tests/src/core/testqgsnetworkaccessmanager.cpp [hunk 1]
```diff
@@ -177,7 +177,7 @@ void TestQgsNetworkAccessManager::initTestCase()
 
   QgsSettings().setValue( QStringLiteral( "/qgis/networkAndProxy/networkTimeout" ), 5000 );
 
-  mHttpBinHost = QStringLiteral( "httpbin.org" );
+  mHttpBinHost = QStringLiteral( "localhost" );
```
Comment (nilsnolde) [inline tests/src/core/testqgsnetworkaccessmanager.cpp:180]: I guess we could add the env var with `localhost` in CI and keep the live service for local tests. But doesn't seem like anyone ran them recently anyways;)

Comment (nyalldawson) [reply tests/src/core/testqgsnetworkaccessmanager.cpp]: Could we do this please? We want to make things as painless as possible to run locally 👍

Comment (nilsnolde) [reply tests/src/core/testqgsnetworkaccessmanager.cpp]: Sure

**Comment by nyalldawson** [comment]
Looking great, thanks!

**Comment by nilsnolde** [comment]
For some reason this is still failing though.. I tried the exact same thing on of our python libs and it worked: https://github.com/gis-ops/routing-py/runs/2253458828?check_suite_focus=true (4. step). I'll try to debug in a small test c++ repo one of these days.

**Comment by nyalldawson** [comment]
@3nids can you see what's missing in the container setup here?

File: tests/src/core/testqgsnetworkaccessmanager.cpp (hunk 1 above)
Comment (nilsnolde) [reply tests/src/core/testqgsnetworkaccessmanager.cpp]: done

**Comment by nilsnolde** [comment]
Added a small test for curling local httpbin early in the pipeline, which worked: https://github.com/qgis/QGIS/pull/42629/checks?check_run_id=2259234336 (4. step right after checkout). I figured I might be able to stop the rest of the test, but apparently no privileges for my own PR (weird github..). 

So it's not a problem with the container setup it seems. Something messing with requests in the test suite?

**Comment by nilsnolde** [comment]
huh.. nothing changed really, but passing now :tada: Wondering if it actually takes the env var though, I forgot to quote the value in the yaml. I'll remove the small curl test tmrw and put it in quotes.

File: tests/src/python/test_qgsblockingnetworkrequest.py [hunk 2]
```diff
@@ -70,7 +70,7 @@ def testFetchBadUrl2(self):
         self.assertIn('Not Found', request.errorMessage())
         reply = request.reply()
         self.assertEqual(reply.error(), QNetworkReply.ContentNotFoundError)
-        self.assertFalse(reply.content())
+        self.assertEqual(reply.content(), '<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN"\n        "http://www.w3.org/TR/html4/strict.dtd">\n<html>\n    <head>\n        <meta http-equiv="Content-Type" content="text/html;charset=utf-8">\n        <title>Error response</title>\n    </head>\n    <body>\n        <h1>Error response</h1>\n        <p>Error code: 404</p>\n        <p>Message: File not found.</p>\n        <p>Error code explanation: HTTPStatus.NOT_FOUND - Nothing matches the given URI.</p>\n    </body>\n</html>\n')
```
Comment (nilsnolde) [inline tests/src/python/test_qgsblockingnetworkrequest.py:73]: In #42630 this test was failing and it actually should've failed when we merged #42553 .. Don't know why it passed and let's see if it fails now..

Comment (nilsnolde) [reply tests/src/python/test_qgsblockingnetworkrequest.py:73]: Turned out some of the local server stuff was refactored and using a mocked server now. Needed to provide that with a response body, works now.

**Comment by nilsnolde** [comment]
I think that's it now!

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**